src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -2,7 +2,7 @@ package is.hail.expr
 
 import is.hail.HailContext
 import is.hail.annotations._
-import is.hail.asm4s.FunctionBuilder
+import is.hail.asm4s.{AsmFunction2, AsmFunction5, FunctionBuilder, GenericTypeInfo, TypeInfo}
```

Comment:
I don't think these are used. I do tend to just do `import is.hail.asm4s._` to indicate to the programmer-reader I'm using the `asm4s` package generally.

Which actually leads me to a code smell, I feel like Relational shouldn't depend on asm4s directly? I think this whole line can be removed now because you've packaged it up in `Emit`.

Reply:
Yes, none were used, removed.  I agree, Relational shouldn't depend on asm4s.  As the code continues to grow (in size and complexity) I think we need to start being more careful about module boundaries and streamlining dependencies.